apps/communications/contacts/test/marionette/lib/views/contact-details/view.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+/* global module */
+
+var ContactsDetailsAccessor = require('./accessors');
+
+function ContactsDetailsView(client, subject) {
+	this.client = client;
+	this.subject = subject;
+	this.accessors = new ContactsDetailsAccessor(client);
+}
+
+ContactsDetailsView.prototype = {
+	makeFavorited: function() {
+		this.accessors.favoritesButton.tap();
+		return this.waitForFavorited;
+	},
+
+	waitForFavorited: function() {
+		function waitFav(temp) {
```

Comment:
We can make this signature more self-documented. How about:
``` js
var self = this;
function waitForHeaderToContainFavoriteClass() {
  return self.detailsHeader.getAttribute('class').indexOf('favorite') != -1;
}
```

`self = this` is a common way to conserve `this` in an inner scope, in JS.

Also, if you think `waitForHeaderToContainFavoriteClass` is too redundant, you can inline this way:
```js
var self = this;
this.client.waitFor(function() {
  return self.detailsHeader.getAttribute('class').indexOf('favorite') != -1;
}
```